### PR TITLE
Docs: Akka Persistence: Better intro for snapshots

### PR DESCRIPTION
**Pull request by sebastianharko**
I believe that this is a slightly better introduction for the section on snapshots since it starts with the WHY (i.e. it gives a motivation for snapshots, even if it's just one sentence).

### CODE REVIEW

File: akka-docs/src/main/paradox/scala/persistence.md [hunk 1]
```diff
@@ -638,7 +638,7 @@ akka.persistence.journal.leveldb.replay-filter {
 <a id="snapshots"></a>
 ## Snapshots
 
-Snapshots can dramatically reduce recovery times of persistent actors and views. The following discusses snapshots
+Some actors in your app may be bound to have extremely long event logs. In that case,  you can use snapshots to reduce recovery times drastically. The following discusses snapshots
 in context of persistent actors but this is also applicable to persistent views.
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/persistence.md]: While you're here could you remove this reference to persistent views? Those disappeared in 2.5 :).

File: akka-docs/src/main/paradox/scala/persistence.md [hunk 2]
```diff
@@ -638,7 +638,7 @@ akka.persistence.journal.leveldb.replay-filter {
 <a id="snapshots"></a>
 ## Snapshots
 
-Snapshots can dramatically reduce recovery times of persistent actors and views. The following discusses snapshots
+Some actors in your app may be bound to have extremely long event logs. In that case,  you can use snapshots to reduce recovery times drastically. The following discusses snapshots
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/persistence.md]: I like giving some more context here. Perhaps even suggest that you might also consider splitting out into many shorter-lived actors?

**Review by raboof — COMMENTED** [review]
Nice improvement, added some more ideas

File: akka-docs/src/main/paradox/scala/persistence.md (hunk 1 above)
Comment (sebastianharko) [inline akka-docs/src/main/paradox/scala/persistence.md]: I removed the reference to persistent views in another PR actually :-)

File: akka-docs/src/main/paradox/scala/persistence.md (hunk 2 above)
Comment (sebastianharko) [inline akka-docs/src/main/paradox/scala/persistence.md]: Yes ! Working on it :-)

**Review by raboof — APPROVED** [review]
Looks reasonable to me, LGTM when jenkins agrees :)

File: akka-docs/src/main/paradox/scala/persistence.md (hunk 2 above)
Comment (sebastianharko) [inline akka-docs/src/main/paradox/scala/persistence.md]: @raboof added some context .. what do you think ? i can add more or less ...

File: akka-docs/src/main/paradox/scala/persistence.md [hunk 3]
```diff
@@ -638,7 +638,7 @@ akka.persistence.journal.leveldb.replay-filter {
 <a id="snapshots"></a>
 ## Snapshots
 
-Snapshots can dramatically reduce recovery times of persistent actors.
+As you model your domain using actors, you may notice that some actors may be prone to accumulating extremely long event logs and experiencing long recovery times. Most often, the right approach is to split out into a set of shorter lived actors. However, when this is not an option, you can use snapshots to reduce recovery times drastically.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/scala/persistence.md]: I don't agree that "most often" the right approach to split out actors with shorter live spans.

Comment (sebastianharko) [inline akka-docs/src/main/paradox/scala/persistence.md]: @johanandren I downgraded "most often" to "sometimes"... just curious, why would not splitting out be the right approach if your domain allows it (for example, splitting out a BankAccount into MonthlyBankAccount) ?

**Review by johanandren — APPROVED** [review]
LGTM, thanks @sebastianharko !

**Pull request merged**